lib/foreman/renderer.rb

```diff
@@ -99,7 +113,10 @@ def snippet(name, options = {})
         begin
           return unattended_render(template, nil, options[:variables] || {})
         rescue => exc
-          raise "The snippet '#{name}' threw an error: #{exc}"
+          # fixes the backtrace but ideally should raise better error message with Foreman rendering exception, same everywhere in this file
```

Comment:
Yes, you're right. I didn't want to do a big refactoring of exceptions in the renderer in this commit but when we do it one day, this should have it's own exception class. If you're OK with introducing SnippetError for this case within this PR, I'm happy to do it. Or I can also just remove the comment, whatever you prefer.